packages/commonwealth/client/scripts/views/SublayoutHeader.tsx

```diff
@@ -36,11 +36,32 @@ type SublayoutHeaderProps = {
 export const SublayoutHeader = ({ onMobile }: SublayoutHeaderProps) => {
   const [isFeedbackModalOpen, setIsFeedbackModalOpen] = useState(false);
   const navigate = useCommonNavigate();
-  const { menuVisible, setMenu, menuName, setMobileMenuName, mobileMenuName } =
-    useSidebarStore();
+  const {
+    menuVisible,
+    setMenu,
+    menuName,
+    setMobileMenuName,
+    mobileMenuName,
+    setUserToggledVisibility,
+    setRecentlyUpdatedVisibility,
+  } = useSidebarStore();
   const { isLoggedIn } = useUserLoggedIn();
   const [isLoginModalOpen, setIsLoginModalOpen] = useState(false);
 
+  useEffect(() => {
+    setRecentlyUpdatedVisibility(menuVisible);
+  }, [menuVisible]);
+
+  function handleToggle() {
+    const isVisible = !menuVisible;
+    const sidebar = document.getElementsByClassName('Sidebar')[0];
+    if (sidebar) sidebar.classList.toggle('onremove', !isVisible);
```

Comment:
accessing html nodes like this, and setting classes like this is totally non-react approach. The whole UI logic should be based on state but now we are bypassing state and modify DOM directly, which is anti-pattern.